File: src/fridaymario/listeners/AllActions.java
package fridaymario.listeners;

import com.intellij.openapi.Disposable;
import com.intellij.openapi.actionSystem.ActionManager;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.actionSystem.ex.AnActionListener;
import org.jetbrains.annotations.NotNull;

public class AllActions implements Restartable {
	private final AnActionListener actionListener;

	public AllActions(final Listener listener) {
		actionListener = new AnActionListener() {
			@Override public void beforeActionPerformed(@NotNull AnAction action,
			                                            @NotNull DataContext dataContext,
			                                            @NotNull AnActionEvent event) {
				listener.onAction(ActionManager.getInstance().getId(action));
			}
		};
	}

	@Override public void start(Disposable disposable) {
		ActionManager.getInstance().addAnActionListener(actionListener);
	}

	@Override public void stop() {
		ActionManager.getInstance().removeAnActionListener(actionListener);
	}

	public interface Listener {
		void onAction(String actionId);
	}
}

File: src/fridaymario/listeners/AllActions.kt
package fridaymario.listeners

import com.intellij.openapi.Disposable
import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.actionSystem.ex.AnActionListener
import fridaymario.util.appMessageBus

class AllActions(private val listener: Listener): Restartable {
    override fun start(disposable: Disposable) {
        appMessageBus.connect(disposable).subscribe(AnActionListener.TOPIC, object: AnActionListener {
            override fun beforeActionPerformed(action: AnAction, dataContext: DataContext, event: AnActionEvent) {
                listener.onAction(ActionManager.getInstance().getId(action))
            }
        })
    }

    interface Listener {
        fun onAction(actionId: String?)
    }
}